### PR TITLE
Rpm Stats Automated Tests

### PR DESCRIPTION
**Pull request by brian-tidepool**
Added tests for PRD and QA2 that filter the clinic patient list, export the RPM stats report, and checks the validity of the data sufficiency column. Automated tests that were implemented were marked in the test strategy spreadsheet:

https://docs.google.com/spreadsheets/d/1gv5q_o5ghR_KMENYvsFmFxGVjKz020O9/edit?usp=sharing&ouid=117042773141856291352&rtpof=true&sd=true

### CODE REVIEW

**Comment by vojtech-cerveny** [comment]
It is kinda hard to read, when this PR contains previous PRs. Should I comment here everything what I commented there or should I compare these 3 PRs? :) 

I would suggest - do a sequential merging - First one is the base, second PR should be into the base (previous) and third PR should be into the second PR (previous). 

* We can see really only changes for that PR (not previous changes)
* You need merge it in sequence. 

Or this should be without previous PRs (so you started from main, not the [clinicianlogin](https://github.com/tidepool-org/WebUITests/tree/clinicianlogin)).

**Comment by brian-tidepool** [comment]
> It is kinda hard to read, when this PR contains previous PRs. Should I comment here everything what I commented there or should I compare these 3 PRs? :)
> 
> I would suggest - do a sequential merging - First one is the base, second PR should be into the base (previous) and third PR should be into the second PR (previous).
> 
> * We can see really only changes for that PR (not previous changes)
> * You need merge it in sequence.
> 
> Or this should be without previous PRs (so you started from main, not the [clinicianlogin](https://github.com/tidepool-org/WebUITests/tree/clinicianlogin)).

The PR depends on the previous PRs. I'll see if i can create a branch that commits on top of merged changes.

**Comment by brian-tidepool** [comment]
@vojtech-cerveny i changed the base branch of the pull request to clinicianlogin. does this separate out the new changes?

**Review by brian-tidepool — COMMENTED** [review]
suppressed notification branch

**Comment by brian-tidepool** [comment]
@vojtech-cerveny i squashed all the new changes related to this branch into one commit: https://github.com/tidepool-org/WebUITests/pull/35/commits/0bcb695149f4f07b6945e39d9d2cf0dec790799f

 let me know if this fixes the readability issue

**Comment by vojtech-cerveny** [comment]
@brian-tidepool - I think that you should have [supressednotificationstest](https://github.com/tidepool-org/WebUITests/tree/supressednotificationstest) as base branch - check this, I tried to change it and it seems that this PR really contains only RPM changes.

Wdyt?

**Review by vojtech-cerveny — CHANGES_REQUESTED** [review]
This needs some extra work. 
Please check also eslint. It throws a lot of errors. 

```
npx eslint . --fix //this automatically fix small issues
npx eslint .

✖ 172 problems (150 errors, 22 warnings)
```

Let me know if you have any questions or I can help you with something.

**Comment by brian-tidepool** [comment]
> @brian-tidepool - I think that you should have [supressednotificationstest](https://github.com/tidepool-org/WebUITests/tree/supressednotificationstest) as base branch - check this, I tried to change it and it seems that this PR really contains only RPM changes.
> 
> Wdyt?

sounds good like a good change, can try this

**Comment by brian-tidepool** [comment]
> This needs some extra work. Please check also eslint. It throws a lot of errors.
> 
> ```
> npx eslint . --fix //this automatically fix small issues
> npx eslint .
> 
> ✖ 172 problems (150 errors, 22 warnings)
> ```
> 
> Let me know if you have any questions or I can help you with something.

all eslint errors fixed except class-methods-use-this which causes an error

see commit:

https://github.com/tidepool-org/WebUITests/commit/2d90b574128c0365533e4bb77da0ed26ad01a532

File: .circleci/config.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@ orbs:
 parameters:
   testEnvironment:
       type: string
-      default: "testParallel"
+      default: "testqa2ChromeRpm"
```
[the remark this answers is not in the document]
Comment (brian-tidepool) [reply .circleci/config.yml]: reverted to testParallel in commit 5002a5489456970f55bbafa3e5e0f46253a10291

File: custom_commands/checkFileContents.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+/* eslint-disable linebreak-style */
+const fs = require('fs');
+const crypto = require('crypto');
+const { hasUncaughtExceptionCaptureCallback } = require('process');
+
+
+module.exports = class checkFileContents {
+  /**
+   * command method of class checkSupressedNotification
+   * @
```
[the remark this answers is not in the document]
Comment (brian-tidepool) [reply custom_commands/checkFileContents.js]: added documentation fileName in commit 7f8d61fe1d3dbc323fccca51dc6b76e8d784338b

File: custom_commands/checkRPMExportSufficiency.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    resolve(arr)
+  })
+  .on('error', (error) => {
+    console.log(error.message);
+    reject(error)
+  });
+  
+  })
+}
+function checkSufficiency(item) {
```
[the remark this answers is not in the document]
Comment (brian-tidepool) [reply custom_commands/checkRPMExportSufficiency.js:30]: added documentation to function in commit bae15b7acf16422b647781bd02916db5c99bb598

File: pageobjects/clinicPatientListPage.js [hunk 4]
```diff
@@ -65,9 +65,301 @@ module.exports = {
           selector: '//button[@id="editClinicProfileSubmit"]',
           locateStrategy: 'xpath',
         },
+        rpmReportButton: {
```
[the remark this answers is not in the document]
Comment (brian-tidepool) [reply pageobjects/clinicPatientListPage.js:68]: deferring the conversion from xpath to css for later time

File: tests/e2e/rpmStatsFunctionalityLastUploadFilterAllRanges.js [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+    const clinicPatientListPage = browser.page.clinicPatientListPage();
+    const clinicPatientList = clinicPatientListPage.section.patientList;
+    let today = moment().format('MM-DD-YYYY')
+    let daysFromToday = 40
+    let startDate = moment().subtract(daysFromToday,"days").format('MMMM D, YYYY')
+    let endDate = moment(startDate,'MMM D, YYYY').add(29,"days").format('MMMM D, YYYY')
+    let startDateFile = moment().subtract(daysFromToday,"days").format('MM-DD-YYYY')
+    let endDateFile = moment(startDateFile,'MM-DD-YYYY').add(29,"days").format('MM-DD-YYYY')
+    let fileName = `RPM Report (${startDateFile} - ${endDateFile}).csv`
+    const attemptsCheckFileExists = 10;
```
[the remark this answers is not in the document]
Comment (brian-tidepool) [reply tests/e2e/rpmStatsFunctionalityLastUploadFilterAllRanges.js:35]: date string creation moved to function createDatesLong and createDateShort in commit 322895bcf7a7f35db7ba1f39879a3619edda6863

File: pageobjects/clinicPatientListPage.js [hunk 6]
```diff
@@ -65,9 +65,301 @@ module.exports = {
...
+          //.moveToElement('@timeInRangeFilterVeryHigh', browser.globals.elementTimeout)
+          .click('@timeInRangeFilterVeryHigh')
+          //.moveToElement('@timeInRangeFilterHigh', browser.globals.elementTimeout)
+          .click('@timeInRangeFilterHigh')
+          .waitForElementVisible('@timeInRangeFilterApply', browser.globals.elementTimeout)
+          .click('@timeInRangeFilterApply');
+        },
+       
+        
+
```
[the remark this answers is not in the document]
Comment (brian-tidepool) [reply pageobjects/clinicPatientListPage.js:344]: steps removed in commit 0f4324d4bc59e927e3a0667ad94ac2f7ee5a42f4

File: tests/e2e/rpmStatsFunctionalityLastUploadFilterToday.js [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+    clinicPatientList.rpmExportDefaultDate();
+    //validate file export results
+    try {
+      const exists = await browser.checkFileExists(attemptsCheckFileExists, fileName);
+      browser.assert.strictEqual(exists, true, 'exported rpm csv file exists ');
+    } catch (error) {
+      console.log(error);
+      browser.assert.strictEqual(false, true, 'exported rpm csv file exists ');
+    }
+    const sufficient = await browser.checkRPMExportSufficiency(filePath);
```
[the remark this answers is not in the document]
Comment (brian-tidepool) [reply tests/e2e/rpmStatsFunctionalityLastUploadFilterToday.js:160]: filePath is the correct variable

**Pull request merged**